### PR TITLE
[bugfix] Properly name PyTorchProfiler traces

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR properly traces generated with PyTorch Profiler.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/profiler/base.py [hunk 1]
```diff
@@ -113,14 +113,20 @@ def _rank_zero_info(self, *args, **kwargs) -> None:
...
-        if self._stage is not None:
-            filename += f"{self._stage}-"
-        filename += str(self.filename)
-        if self._local_rank is not None:
-            filename += f"-{self._local_rank}"
-        filename += extension
+        filename = self._add_suffix(filename, self._stage.value if self._stage else None)
+        filename = self._add_suffix(filename, self.filename if self.filename else None)
+        filename = self._add_suffix(filename, str(self._local_rank) if self._local_rank is not None else None)
+        filename += extension if extension else ''
```
Comment (carmocca) [inline pytorch_lightning/profiler/base.py:129]: why the if here? It should be always a str

```suggestion
        filename += extension
```

File: pytorch_lightning/profiler/base.py [hunk 2]
```diff
@@ -113,14 +113,20 @@ def _rank_zero_info(self, *args, **kwargs) -> None:
...
         filename = ""
-        if self._stage is not None:
-            filename += f"{self._stage}-"
-        filename += str(self.filename)
-        if self._local_rank is not None:
-            filename += f"-{self._local_rank}"
-        filename += extension
+        filename = self._add_suffix(filename, self._stage.value if self._stage else None)
+        filename = self._add_suffix(filename, self.filename if self.filename else None)
+        filename = self._add_suffix(filename, str(self._local_rank) if self._local_rank is not None else None)
```
Comment (carmocca) [inline pytorch_lightning/profiler/base.py:128]: What's the advantage of all this? If I'm not missing something else it just needs to change

```python
        if self._stage is not None:
            filename += f"-{self._stage}"
```

or  if you want to keep the previous format

```python
        if self._stage is not None:
            filename = f"{self._stage}-" + filename
```

File: pytorch_lightning/profiler/pytorch.py [hunk 3]
```diff
@@ -426,12 +426,13 @@ def stop(self, action_name: str) -> None:
 
             def on_trace_ready(profiler):
                 if self.dirpath is not None:
+                    filename = self._prepare_filename(extension="")
                     if self._export_to_chrome:
-                        handler = tensorboard_trace_handler(self.dirpath, self._prepare_filename(extension=""))
+                        handler = tensorboard_trace_handler(self.dirpath, filename + "-" + str(action_name))
```
Comment (carmocca) [inline pytorch_lightning/profiler/pytorch.py:431]: Why not add the `action_name` keyword argument to `prepare_filename`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -264,6 +264,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed `BaseFinetuning` callback to properly handle parent modules w/ parameters ([#7931](https://github.com/PyTorchLightning/pytorch-lightning/pull/7931))
 
 
+- Fixed `PyTorchProfiler` generated filenames ([#8009](https://github.com/PyTorchLightning/pytorch-lightning/pull/8009))
```
Comment (carmocca) [inline CHANGELOG.md:267]: Can you elaborate on what's the fix exactly? Or what was broken

**Review by carmocca — CHANGES_REQUESTED** [review]

File: pytorch_lightning/profiler/base.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/profiler/base.py:129]: Done !

File: CHANGELOG.md (hunk 4 above)
Comment (tchaton) [reply CHANGELOG.md:267]: Hey @carmocca,
The filenames looked like fit-None.{some_number} and the action name for PyTorch Profiler wasn't saved, creating overlap in TensorBoard Plugin.

File: pytorch_lightning/profiler/pytorch.py (hunk 3 above)
Comment (tchaton) [reply pytorch_lightning/profiler/pytorch.py:431]: Done !

**Review by carmocca — APPROVED** [review]

**Pull request merged**